### ISSUE TITLE
Cross-version conflict

### ISSUE THREAD

**Issue by rtimush** [netvl/picopickle#3]
Picopicle depends on `shapeless_2.10.4` despite of the real scala version used.

**build.sbt**

``` scala
scalaVersion := "2.10.5"

addCompilerPlugin("org.scalamacros" %% "paradise" % "2.0.1" cross CrossVersion.full)

libraryDependencies ++= Seq(
    "io.github.netvl.picopickle" %% "picopickle-core" % "0.1.1",
    "com.chuusai" %  "shapeless" % "2.0.0" cross CrossVersion.full
)
```

**Error:**

```
$ sbt update
[info] Loading global plugins from /Users/roman/.sbt/0.13/plugins
[info] Set current project to bug (in build file:/private/tmp/bug/)
[info] Updating {file:/private/tmp/bug/}bug...
[info] Resolving org.fusesource.jansi#jansi;1.4 ...
[info] Done updating.
[error] Modules were resolved with conflicting cross-version suffixes in {file:/private/tmp/bug/}bug:
[error]    com.chuusai:shapeless _2.10.4, _2.10.5
```

**Comment by netvl** [comment]
Yes, picopickle intentionally depended on the 2.10.4 version because shapeless uses full binary version suffixes and shapeless 2.1.0 was not available for 2.10.5 until the end of March. I'll update the version ASAP.

**Comment by netvl** [comment]
Fixed in 0.1.2. It will be in Maven Central soon.
